### PR TITLE
Fix complex beam handling

### PR DESCRIPTION
**Pull request by philbull**
* Fixes to enable complex beam handling, including in pixelised beam interpolation
* Improved floating point handling near poles etc.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[flake8]
```
Comment (steven-murray) [inline .flake8:1]: We are currently setting configuration for flake8 in the `setup.cfg`. You're more than welcome to copy/paste that config into the `.flake8` file if you prefer it there (I think I do, actually).

File: src/vis_cpu/conversions.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def enu_to_az_za(enu_e, enu_n, orientation="astropy", periodic_azimuth=True):
     ], "orientation must be either 'astropy' or 'uvbeam'"
 
     lsqr = enu_n ** 2.0 + enu_e ** 2.0
-    zeta = np.where(lsqr < 1.0, np.sqrt(1.0 - lsqr), 0.0)
+    zeta = np.where(lsqr < 1.0, np.sqrt(np.abs(1.0 - lsqr)), 0.0)
```
Comment (steven-murray) [inline src/vis_cpu/conversions.py:53]: Hmmm, for speed I guess just creating a mask array is going to be more efficient (and avoid the spurious warnings), i.e.

```
mask = lsqr < 1
zeta = np.zeros_like(lsqr)
zeta[mask] = np.sqrt(1 - lsqr[mask])
```
It's a few more lines, but avoids the extra calculations where the mask isn't true.

**Review by steven-murray — COMMENTED** [review]
Thanks @philbull, these kinds of checks are incredibly useful. Just a couple of quick comments then should be an easy merge. 

The deploy check is being fixed in #19 , which should be merged soon.

**Comment by steven-murray** [comment]
Ah, also update the changelog!

File: src/vis_cpu/conversions.py (hunk 2 above)
Comment (philbull) [reply src/vis_cpu/conversions.py:53]: Thanks, I've made this change.

File: .flake8 (hunk 1 above)
Comment (philbull) [reply .flake8:1]: Done, thanks for pointing that out.

**Review by steven-murray — APPROVED** [review]
Thanks @philbull, looks ready to go!

**Pull request merged**